pkg/apiserver/resthandler.go

```diff
@@ -347,6 +347,21 @@ func DeleteResource(r RESTDeleter, ctxFn ContextFunc, namer ScopeNamer, codec ru
 			ctx = api.WithNamespace(ctx, namespace)
 		}
 
+		options := &api.DeleteOptions{}
+		if checkBody {
+			body, err := readBody(req.Request)
+			if err != nil {
+				errorJSON(err, codec, w)
+				return
+			}
+			if len(body) > 0 {
```

Comment:
This is partially to distinguish between old behavior and new behavior (nil to rest storage means old behavior, ie delete immediately).  That might not be required anymore, but a few places pass nil to a storage object instead of each spot trying to load it's own default.  Agree it's non-obvious as is.


> On Mar 18, 2015, at 8:48 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/apiserver/resthandler.go:
> 
> > @@ -347,6 +347,21 @@ func DeleteResource(r RESTDeleter, ctxFn ContextFunc, namer ScopeNamer, codec ru
> >  			ctx = api.WithNamespace(ctx, namespace)
> >  		}
> >  
> > +		options := &api.DeleteOptions{}
> > +		if checkBody {
> > +			body, err := readBody(req.Request)
> > +			if err != nil {
> > +				errorJSON(err, codec, w)
> > +				return
> > +			}
> > +			if len(body) > 0 {
> It might make sense to create options if none were provided and then run them through the defaulting pass. Otherwise, code all over will presumably need to reason about default behavior in the absence of options.
> 
> —
> Reply to this email directly or view it on GitHub.
>